server/api/images.js

```diff
@@ -0,0 +1,14 @@
+// @flow
+import Router from 'koa-router';
+import auth from '../middlewares/authentication';
+import { getSignedImageUrl } from '../utils/s3';
+
+const router = new Router();
+
+router.post('images.info', auth(), async ctx => {
```

Comment:
I'm actually unclear how the auth is succeeding here, I haven't ran this branch locally yet – but it seems like no auth would be available without frontend changes?

Reply:
Thinking about the above suggestion some more it would also help with https://github.com/outline/outline/issues/1010 and form the foundation for file attachments. 

I can go ahead and do this work in a separate branch if you like.

Reply:
The auth is going through because the image is being requested from the same origin, so all the session cookies etc are going through with the request.

I was going to say actually that having images registered in a table could go towards the file attachment issues I've seen floating about. You could have all uploaded "assets" registered in the new table, and all permission checks, cleanup sweeps, audits etc can happen from that. My questions would be:
- in light of the suggestion to have outline be private by default, should all image urls in the markdown be saved as `/image.info?id=` regardless, and then the proxy can just choose to redirect with the signed url or the public url depending on the setting? (even so, people may as well just let the signed urls happen anyway, unless as you say s3 is not being used)
- doing this does widen the scope of this PR, so how'd you want to proceed? If you think this warrants all going out at once, and not just doing the "simple" permissionsless auth check, then perhaps I can implement/address the rest of the comments here and then it can be merged this into a new branch rather than master to be the actual feature branch for private content.

Reply:
> in light of the suggestion to have outline be private by default, should all image urls in the markdown be saved as /image.info?id= regardless

I think I'd prefer that, it should be less edge-cases to deal with.

Reply:
> doing this does widen the scope of this PR, so how'd you want to proceed?

My plan is to start a separate PR with database storage of the image attachments, I'll add you to the review process – once that's approved and on master you can pull into this branch and it should be a relatively simple change to use the new attachment id instead of the path as the key and authenticate access.

Reply:
PR: https://github.com/outline/outline/pull/1144